cmd/argot/main.go

```diff
@@ -158,12 +158,19 @@ func main() {
 			argotunnel.EnableMetrics(5 * time.Second)
 			argotunnel.SetVersion(version)
 
-			ctx, cancel := context.WithCancel(context.Background())
-			argo := argotunnel.NewController(kclient, log,
+			options := []argotunnel.Option{
 				argotunnel.IngressClass(*ingressclass),
-				argotunnel.Secret(originsecret.Name, originsecret.Namespace),
 				argotunnel.ResyncPeriod(*resyncperiod),
 				argotunnel.Workers(*workers),
+			}
+
+			for _, obj := range *originsecrets {
+				options = append(options, argotunnel.Secret(obj.Name, obj.Namespace))
```

Comment:
what you'll start to notice here is that this command-line flag variant of default secrets will start to expose the same/similar (its small scope, but approximately the same) issues as the label variant.